components/style/properties/data.py

```diff
@@ -78,6 +82,12 @@ def __init__(self, style_struct, name, animatable=None, derived_from=None, keywo
         self.depend_on_viewport_size = depend_on_viewport_size
         self.derived_from = (derived_from or "").split()
         self.complex_color = complex_color
+        self.cast_type = cast_type
+
+        # This is a workaround for gecko constants with unexpected names i.e., a constant
+        # corresponding to a keyword 'foo-bar' declared as FOOBAR instead of FOO_BAR
+        if self.keyword is not None:
+            self.keyword.ignore_hyphens = ignore_hyphens
```

Comment:
IMO workarounds like `ignore_hyphens` that get used just once probably shouldn't exist 😄 I'd prefer if we manually did image-rendering. Though I'm good with the map solution too.